keras_cv/layers/preprocessing/resizing.py

```diff
@@ -0,0 +1,282 @@
+
+    Args:
+        height: Integer, the height of the output shape.
+        width: Integer, the width of the output shape.
+        interpolation: String, the interpolation method. Defaults to `"bilinear"`.
+            Supports `"bilinear"`, `"nearest"`, `"bicubic"`, `"area"`, `"lanczos3"`,
+            `"lanczos5"`, `"gaussian"`, `"mitchellcubic"`.
+        crop_to_aspect_ratio: If True, resize the images without aspect
+            ratio distortion. When the original aspect ratio differs from the target
+            aspect ratio, the output image will be cropped so as to return the
+            largest possible window in the image (of size `(height, width)`) that
+            matches the target aspect ratio. By default
+            (`crop_to_aspect_ratio=False`), aspect ratio may not be preserved.
+        pad_to_aspect_ratio:
+        bounding_box_format:
+    """
+
+    def __init__(
+        self,
+        height,
+        width,
+        interpolation="bilinear",
+        crop_to_aspect_ratio=False,
+        pad_to_aspect_ratio=False,
+        bounding_box_format=None,
+        **kwargs,
+    ):
+        self.height = height
+        self.width = width
+        self.interpolation = interpolation
+        self.crop_to_aspect_ratio = crop_to_aspect_ratio
+        self.pad_to_aspect_ratio = pad_to_aspect_ratio
+        self._interpolation_method = keras_cv.utils.get_interpolation(interpolation)
+        self.bounding_box_format = bounding_box_format
+
+        if pad_to_aspect_ratio and crop_to_aspect_ratio:
+            raise ValueError(
+                "`Resizing()` expects at most one of `crop_to_aspect_ratio` or "
+                "`pad_to_aspect_ratio` to be True."
+            )
+
+        if not pad_to_aspect_ratio and bounding_box_format:
+            raise ValueError(
+                "Resizing() only supports bounding boxes when in "
+                "`pad_to_aspect_ratio=True` model.  "
+                "Please pass `pad_to_aspect_ratio=True`"
+                "when processing bounding boxes with `Resizing()`"
+            )
+        super().__init__(**kwargs)
+
+    def _augment(self, inputs):
+        images = inputs.get("images", None)
+        bounding_boxes = inputs.get("bounding_boxes", None)
+
+        if images is not None:
+            images = tf.expand_dims(images, axis=0)
+            inputs["images"] = images
+
+        if bounding_boxes is not None:
+            bounding_boxes = tf.expand_dims(bounding_boxes, axis=0)
+            inputs["bounding_boxes"] = bounding_boxes
+
+        outputs = self._batch_augment(inputs)
+
+        if images is not None:
+            images = tf.squeeze(outputs["images"], axis=0)
+            inputs["images"] = images
+
+        if bounding_boxes is not None:
+            bounding_boxes = tf.squeeze(outputs["bounding_boxes"], axis=0)
+            inputs["bounding_boxes"] = bounding_boxes
+        return inputs
```

Comment:
I think in its current form this will pass through any keys included in `inputs` that aren't augmented (e.g. segmentation masks).

We should either:
- Throw if segmasks (or other unsupported data) are passed in
- Drop unsupported data from the output
- Add support for segmasks (and any other required data keys for our KPLs)

This is one benefit that we lose by overriding the BaseImageAugmentationLayer's `augment` method, but if it's necessary let's at least make sure we don't silently pass through broken segmentation masks.

Reply:
Do we want to support segmasks before 0.4? This would be helpful for DeepLab

Reply:
That would be great but lets keep this PR localized.  I'll raise an error if any key other than `['images', 'labels', 'targets', 'bounding_boxes']` is passed for now.

Reply:
raising error in this PR sounds good -- can you also add a TODO here as well?